octobot_trading/exchanges/connectors/ccxt_exchange.py

```diff
@@ -271,124 +308,699 @@ async def get_order_book(self, symbol: str, limit: int = 5, **kwargs: dict) -> t
+            self.logger.exception(e, True, "Failed to fetch open stop orders using"
+                                           " get_open_stop_order_using_stop_loss_endpoint")
+            return []
+
+    async def get_closed_orders(self, symbol: str = None, since: int = None, limit: int = None,
+                                check_completeness: bool = True, **kwargs: dict) -> list:
+        """
+            all known get_closed_orders methods should be added here so untested exchanges have higher chance of success
+        """
+        limit = self.cut_order_pagination_limit(limit)
+        defined_methods = self.connector_config.GET_CLOSED_ORDERS_METHODS
+        orders = []
+        if self.get_closed_orders_default.__name__ in defined_methods:
+            orders += await self.get_closed_orders_default(symbol, since, limit, check_completeness=check_completeness,
+                                                           **kwargs)
+        if self.get_closed_stop_orders_using_stop_loss_params.__name__ in defined_methods:
+            orders += await self.get_closed_stop_orders_using_stop_loss_params(symbol, since, limit,
+                                                                               check_completeness=check_completeness,
+                                                                               **kwargs)
+        return orders
+
+    async def get_closed_orders_default(self, symbol: str = None, since: int = None, limit: int = None,
+                                        check_completeness: bool = True, **kwargs: dict) -> list:
+        if self.client.has.get('fetchClosedOrders'):
             with self.error_describer():
-                return await self.client.fetch_closed_orders(symbol=symbol, since=since, limit=limit, params=kwargs)
+                raw_order = await self.client.fetch_closed_orders(
+                    symbol=symbol, since=since, limit=limit, params=kwargs)
+                return await self.exchange_manager.exchange.parse_orders(raw_order,
+                                                                         check_completeness=check_completeness)
         else:
             raise octobot_trading.errors.NotSupported("This exchange doesn't support fetchClosedOrders")
 
+    async def get_closed_stop_orders_using_stop_loss_params(self, symbol, since, limit,
+                                                            check_completeness: bool = True, **kwargs) -> list:
+        try:
+            if kwargs := self.exchange_manager.exchange.custom_get_closed_orders_stop_params(kwargs):
+                orders = await self.get_closed_orders_default(symbol=symbol, since=since, limit=limit,
+                                                              check_completeness=check_completeness, **kwargs)
+                return orders
+        except ccxt.AuthenticationError as e:
+            self.logger.debug(f"(known issue) Fail to fetching closed stop orders : {e}")
```

Comment:
"known issue"?

Reply:
oops I copied that from somewhere. I removed it